newIDE/app/src/Profile/Subscription/PlanCard.js

```diff
@@ -14,59 +18,197 @@ import { useResponsiveWindowWidth } from '../../UI/Reponsive/ResponsiveWindowMea
+              months
+            </Trans>
+          </Text>
+        );
+      }
+    }
+  } else {
+    if (price.periodCount === 1) {
+      if (price.isPerUser) {
+        return (
+          <Text key="year" noMargin color="secondary">
+            <Trans>
+              {price.amountInCents / 100}
+              {price.currency === 'EUR' ? '€' : '$'}/seat/year
+            </Trans>
+          </Text>
+        );
+      } else {
+        return (
+          <Text key="year" noMargin color="secondary">
+            <Trans>
+              {price.amountInCents / 100}
+              {price.currency === 'EUR' ? '€' : '$'}/year
+            </Trans>
+          </Text>
+        );
+      }
+    } else {
+      if (price.isPerUser) {
+        return (
+          <Text key="year" noMargin color="secondary">
+            <Trans>
+              {price.amountInCents / 100}
+              {price.currency === 'EUR' ? '€' : '$'}/seat every{' '}
+              {price.periodCount} years
+            </Trans>
+          </Text>
+        );
+      } else {
+        return (
+          <Text key="year" noMargin color="secondary">
+            <Trans>
+              {price.amountInCents / 100}
+              {price.currency === 'EUR' ? '€' : '$'} every {price.periodCount}{' '}
+              years
+            </Trans>
+          </Text>
+        );
+      }
+    }
+  }
+};
+
+/**
+ * If price is given, it will be displayed. Otherwise, the prices stored in the
+ * subscription plan will be displayed.
+ */
+const getPlanPrices = ({
+  prices,
+  price,
   hidePrice,
 }: {
-  plan: PlanDetails,
+  prices: ?(SubscriptionPlanPrice[]),
+  price: ?SubscriptionPlanPrice,
   hidePrice?: boolean,
 }): React.Node => {
-  if (hidePrice || plan.monthlyPriceInEuros === null) return null;
-  if (plan.monthlyPriceInEuros === 0)
-    return (
-      <Text noMargin color="secondary">
-        <Trans>Free</Trans>
-      </Text>
-    );
-  const prices = [];
-  prices.push(
-    plan.isPerUser ? (
-      <Text key="month" noMargin color="secondary">
-        <Trans>{plan.monthlyPriceInEuros}€/seat/month</Trans>
-      </Text>
-    ) : (
-      <Text key="month" noMargin color="secondary">
-        <Trans>{plan.monthlyPriceInEuros}€/month</Trans>
-      </Text>
-    )
-  );
-  if (plan.yearlyPriceInEuros) {
-    prices.push(
-      plan.isPerUser ? (
-        <Text key="year" noMargin color="secondary">
-          <Trans>or {plan.yearlyPriceInEuros}€/seat/year</Trans>
-        </Text>
-      ) : (
-        <Text key="year" noMargin color="secondary">
-          <Trans>or {plan.yearlyPriceInEuros}€/year</Trans>
-        </Text>
-      )
-    );
+  if (hidePrice) return null;
+  if (price) {
+    return getPlanPrice(price);
+  }
+
+  if (prices && prices.length > 0) {
+    const displayedPrices = prices
+      .sort((priceA, priceB) => {
+        // TODO: Decide if the front end should sort the prices or if it should respect
```

Comment:
I believe it should respect the backend, and have the backend sort it. Otherwise we're stuck with a release if we want to make a change